Adding Protobufs for CorfuStore [PART 1]

zalokhan opened the pull request:
## Overview

Description:
Introducing the relevant protobufs required to initialize the CorfuStore.

Why should this be merged: 
Introducing CorfuStore.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

pankti-m commented on runtime/proto/corfu_options.proto:
```diff
@@ -0,0 +1,18 @@
+syntax = "proto2";
```
why proto2 and not 3?

xiaoqin2012 commented on test/src/test/resources/proto/test_schema.proto:
```diff
@@ -0,0 +1,24 @@
+syntax = "proto2";
+
```
Same here, why someplace use proto3 some place uses proto2?

zalokhan commented:
@pankti-m @xiaoqin2012 Addressed your comments.
Changed all Corfu options and metadata to use proto3.
The test protobuf classes will continue to use proto2 to test the backward compatibility.

lixinchengdu commented on runtime/proto/corfu_options.proto:
```diff
@@ -0,0 +1,18 @@
+}
+
+extend google.protobuf.FieldOptions {
+    SchemaOptions schema = 1036;
```
Can someone explain to me what the magic number 1036 means here?

hisundar replied:
It is google protobuf's FieldOptions field number that does not collide with an existing number.

lixinchengdu commented on test/src/test/resources/proto/test_schema.proto:
```diff
@@ -0,0 +1,24 @@
+import "corfu_options.proto";
+
+message Uuid {
+    optional uint64 msb = 1;
```
Can I know why the both msb and lsb are optional? Does it means that if one field is missing, the value of that filed is 0?

hisundar replied:
"optional" is the default specifier in proto3 
In this case it should really be "required" but this test is highlighting the possibility of defining protobufs in this manner.

medhavidhawan commented on runtime/proto/corfu_options.proto:
```diff
@@ -0,0 +1,18 @@
+}
+
+extend google.protobuf.FieldOptions {
+    SchemaOptions schema = 1036;
```
please add a comment

zalokhan replied:
Done.

medhavidhawan commented on runtime/proto/corfu_store_metadata.proto:
```diff
@@ -0,0 +1,38 @@
+message TableMetadata {
+    TableName table_name = 1;
+    bool disk_based = 2;
+    bool cache = 3;
```
please add comment for this

zalokhan replied:
Done.
